dotnet/src/Extensions/Extensions.UnitTests/TemplateEngine/Prompt/PromptTemplateEngineTests.cs

```diff
@@ -149,16 +151,18 @@ string MyFunctionAsync(SKContext context)
             return $"F({context.Variables.Input})";
         }
 
-        ISKFunction func = SKFunction.FromNativeMethod(Method(MyFunctionAsync), this);
-        Assert.NotNull(func);
+        List<ISKFunction> functions = new()
+        {
+            SKFunction.FromNativeMethod(Method(MyFunctionAsync), this),
+        };
+
+        Assert.NotNull(functions[0]);
```

Comment:
Looks like this change is present in a couple of places in tests (replacing one function with list of functions), but we still work only with one function and access it by index `functions[0]`. Is this change required? I think if we revert it, it should still work, but maybe I'm missing something, please let me know.

Reply:
The previous implementation breaks when using the Invoke native reflection. 
Says that the target is not the same.

Not sure if this is a .Net bug or something, but when I use within a `List` it works.

@stephentoub do you have any insights what might be the problem here?

Reply:
Interestingly enough, I retried with the previous logic and it worked. Not sure what might have happened in my last experiment, seems to be related to local execution. Still not sure what was it but can't replicate it now.